### DISCUSSION TITLE
trying to get SSO to work

### DISCUSSION THREAD

**Discussion by aikijitsu** [argoproj/argo-cd#12494] [Q&A]
I followed the oidc instructions but don't see a sso login button on the login page. What should I look for?
Or Dex. Just trying to get SSO to work.

**Comment by larsWtz** [comment]
@aikijitsu can you share your config? Don't forget to mask credentials if used. I just implemented SSO with Azure AD in a project using it as the OIDC provider.

**Comment by aikijitsu** [comment]
the indentation isnt showing up when I paste but here are the 3 files. thanks for looking

apiVersion: v1
kind: ConfigMap
metadata:
name: argocd-rbac-cm
labels:
app.kubernetes.io/name: argocd-rbac-cm
app.kubernetes.io/part-of: argocd

data:
    policy.default: role:readonly
    policy.csv: |
    p, role:org-admin, applications, *, /, allow
    p, role:org-admin, clusters, get, *, allow
    p, role:org-admin, repositories, get, *, allow
    p, role:org-admin, repositories, create, *, allow
    p, role:org-admin, repositories, update, *, allow
    p, role:org-admin, repositories, delete, *, allow
    g, "object id of admin group", role:org-admin

apiVersion: v1
kind: Secret
metadata:
name: argocd-secret
labels:
app.kubernetes.io/name: argocd-secret
app.kubernetes.io/part-of: argocd
type: Opaque

data:
oidc.azure.clientSecret: secret from appregistration base 64 encoded

apiVersion: v1
kind: ConfigMap
metadata:
name: argocd-cm
labels:
app.kubernetes.io/name: argocd-cm
app.kubernetes.io/part-of: argocd
data:
url: https://myipaddress
oidc.config: |
name: Azure
issuer: https://login.microsoftonline.com/mytenantid/v2.0
clientID: application(clientId) from registration
clientSecret: $oidc.azure.clientSecret
requestedIDTokenClaims:
groups:
essential: true
requestedScopes:
- openid
- profile
- email

**Comment by larsWtz** [comment]
Looks pretty similar to my config, i could imagine that the URL won't work like that, that's the URL you get redirected to with the /login?...
but for the config it actually only wants the base URL, so i have a DNS entry like https://my-dns.com as the url. Also you can check with a tool like lens or with kubectl describe cm argocd-cm -n namespace, if your changes are actually currently deployed in your cluster.

Last but not least make sure to get a fresh argocd-server pod so it starts up with the current changes in your config maps. With a kubectl delete pod nameOfArgoCdServerPod or also through lens UI.

**Reply by aikijitsu** [reply]
I changed the url to just the IP and it still, isn't working I guess I could create a dns record. What I have been doing is make the change then run kubectl apply f ./argocd-cm.yaml also for the other files same process, then I go and delete all the running pods from the portal.

**Reply by larsWtz** [reply]
In AzureAD you have the following configured?
The Application adding Web Platform with redirect URI https://myipaddress/auth/callback
and adding Desktop/Mobile Application with redirect URI http://localhost:8085/auth/callback

**Reply by aikijitsu** [reply]
Both of those are there. Also I added sign on url with the ip.

**Reply by aikijitsu** [reply]
I started with 
Configure Argo to use the new Azure AD App registration
https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/microsoft/

**Reply by larsWtz** [reply]
Do you see anything in the logs of the argo-cd-server pod log regarding the initializing of the OIDC provider?

on log level INFO, there should be a message for that,
level=info msg="Initializing OIDC provider (issuer: https://login.microsoftonline.com/tenant-id/v2.0)"
level=info msg="OIDC supported scopes: [openid profile email offline_access]"

**Reply by aikijitsu** [reply]
I don't see anything related to OIDC. Below is repeatedly logged 

time="2023-02-23T13:45:58Z" level=info msg="received unary call /session.SessionService/GetUserInfo" grpc.method=GetUserInfo grpc.request.content= grpc.service=session.SessionService grpc.start_time="2023-02-23T13:45:58Z" span.kind=server system=grpc

time="2023-02-23T13:45:57Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=Get grpc.service=cluster.SettingsService grpc.start_time="2023-02-23T13:45:57Z" grpc.time_ms=9.654 span.kind=server system=grpc

**Comment by aikijitsu** [comment]
![image](https://user-images.githubusercontent.com/78279931/223868070-e4308ac2-ac28-4d7d-8eff-120db3e39a8c.png)
![image](https://user-images.githubusercontent.com/78279931/223868093-739ae5d8-15d4-4059-aad5-25b87e87a87a.png)
![image](https://user-images.githubusercontent.com/78279931/223868111-775891d8-2c68-43fd-8a26-4c2795f1e1db.png)
